extensions/WindowsSandboxExtension/Providers/WindowsSandboxComputeSystem.cs

```diff
@@ -0,0 +1,204 @@
+                    if (_windowsSandboxExeProcess == null || _windowsSandboxExeProcess.HasExited)
+                    {
+                        var system32Path = Environment.GetFolderPath(Environment.SpecialFolder.System);
+                        var windowsSandboxExePath = Path.Combine(system32Path, Constants.WindowsSandboxExe);
+
+                        _windowsSandboxExeProcess = new();
+                        _windowsSandboxExeProcess.StartInfo.FileName = windowsSandboxExePath;
+                        _windowsSandboxExeProcess.EnableRaisingEvents = true;
+                        _windowsSandboxExeProcess.Exited += WindowsSandboxProcessExited;
+
+                        Thread.Sleep((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
+
+                        State = ComputeSystemState.Running;
+                        TraceLogging.StartingWindowsSandbox();
+
+                        _windowsSandboxExeProcess.Start();
+
+                        PInvoke.SetForegroundWindow((HWND)_windowsSandboxExeProcess.MainWindowHandle);
+                    }
+
+                    BringWindowsSandboxClientToForeground();
+
+                    return new ComputeSystemOperationResult();
+                }
+            }
+            catch (Exception ex)
+            {
+                _log.Error(ex, "Failed to start Windows Sandbox");
+                TraceLogging.ExceptionThrown(ex);
+                return new ComputeSystemOperationResult(ex, Resources.GetResource("WindowsSandboxFailedToStart", _log), "Failed to start Windows Sandbox");
+            }
+        }).AsAsyncOperation();
+    }
+
+    private void WindowsSandboxProcessExited(object? sender, EventArgs e)
+    {
+        State = ComputeSystemState.Stopped;
+        _windowsSandboxExeProcess?.Dispose();
+        _windowsSandboxExeProcess = null;
+    }
+
+    private Process? WindowsSandboxClientProcess()
+    {
+        return Process.GetProcessesByName("WindowsSandboxClient").FirstOrDefault();
+    }
+
+    private void BringWindowsSandboxClientToForeground()
+    {
+        var clientProcess = WindowsSandboxClientProcess();
+        var windowHandle = clientProcess?.MainWindowHandle ?? IntPtr.Zero;
+
+        PInvoke.SetForegroundWindow((HWND)windowHandle);
+    }
+
+    public IAsyncOperation<ComputeSystemOperationResult> TerminateAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            lock (_windowsSandboxStartLock)
+            {
+                if (_windowsSandboxExeProcess != null)
+                {
+                    _windowsSandboxExeProcess.CloseMainWindow();
+                    WindowsSandboxClientProcess()?.CloseMainWindow();
+                }
+
+                return new ComputeSystemOperationResult();
+            }
+        }).AsAsyncOperation();
+    }
+
+    public IApplyConfigurationOperation CreateApplyConfigurationOperation(string configuration) => throw new NotImplementedException();
```

Comment:
For all your extension methods please return the appropriate *Result type or null if the method returns an interface. This is because since this is cross process communication COM will stomp out the exception and the exception Dev Home will receive will be a generic COM exception. 

here is an example of how you can tell provide Dev Home with error text that you don't implement the methods:

https://github.com/microsoft/devhome/blob/2a36c1aefa27e2bb6a33b145d0e5d6876f10c12d/HyperVExtension/src/HyperVExtension/Providers/HyperVProvider.cs#L90

Reply:
What about the methods that expect an interface like `CreateApplyConfiguration` ? There is no ApplyConfigurationOperation object in the SDK. Should I create one just to return an exception?

Reply:
for those you'd return null. We see those as failures as well. Thats how I handle it in this method:
https://github.com/microsoft/devhome/blob/8a35d2f9d6f30164268afa0760f514dd01e32802/HyperVExtension/src/HyperVExtension/Providers/HyperVProvider.cs#L98